### PR TITLE
External app pre-module registration hook

### PR DESCRIPTION
**Pull request by ResearchDaniel**
Provide a hook for external applications such that they can perform the same setup as the internal apps.

### CODE REVIEW

**Comment by petersteneteg** [comment]
Any reason why you can not just rise a cmake error in your external app if the module is missing, and ask the user to enable it? 
I would rather that we make it possible to add inviwo as a subproject, than that we add more complexity for including external projects like this

**Comment by ResearchDaniel** [comment]
Convenience and consistency with the “internal” code was the reason

**Comment by petersteneteg** [comment]
Mmm I can understand that. But Im trying to reduce the complexity of our cmake setup, and this would go in the opposite direction of that...

File: CMakeLists.txt [hunk 1]
```diff
@@ -125,7 +128,7 @@ endif()
 ivw_register_modules(all_modules)            # Add inviwo modules
 add_subdirectory(src/qt)                     # Add the Qt network editor
 add_subdirectory(apps)                       # Add all the applications, uses the modules.
-ivw_add_external_projects()                  # Add external projects
+ivw_add_external_apps()                      # Add external projects
```
Comment (petersteneteg) [inline CMakeLists.txt:131]: this is not just for apps, way rename it? it also makes it inconsistent with the cmake flag use and the comments...

**Review by petersteneteg — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (ResearchDaniel) [reply CMakeLists.txt:131]: I’ve only seen it used for apps and thought it would be more consistent with the other names (IVW_APP).

**Comment by ResearchDaniel** [comment]
Ok, let’s skip this PR then and see what should be done once the complexity is reduced.

**Comment by petersteneteg** [comment]
reducing complexity is more of a long term goal... but ideally your app should be able to consume inviwo as a dependency not make it self a part of it....

**Pull request closed without merging**